### PR TITLE
UI Tests Using Espresso

### PR DESCRIPTION
**Pull request by abs51295**
@chhavip I would update this PR with commits for each activity test.

**Tests added:**

1. Device State Tests - Run this tests first on your device before running any tests to check if animators, animations, and window transitions are disabled.

**How to Run All the Tests?**

1. Connect a physical device or emulator before running any kind of tests. 
2. After that, there are two ways to run the tests:

- If you want to run all of the tests within a given test class, press the run icon as highlighted below:

testall
     
    
-  If you want to test individual functions within a test class, press the run icon next to the functions annotated with `@Test` annotation as highlighted below:

testsingle

3. After running the tests you can see the test results in the window that is opened in Android Studio:

testresults

### CODE REVIEW

**Comment by chhavip** [comment]
@abs51295 Add instructions to run the tests in the description

File: app/src/main/java/com/systers/conference/login/LoginActivity.java [hunk 1]
```diff
@@ -137,7 +137,7 @@ public void onCompleted(JSONObject object, GraphResponse response) {
     }
 
     private void startRegisterActivity() {
-        AccountUtils.setLoginVisited(this);
+        AccountUtils.setLoginVisited(this, true);
```
Comment (chhavip) [inline app/src/main/java/com/systers/conference/login/LoginActivity.java]: why are these changes added to this PR? Does not concern testing in my opinion

File: app/src/main/java/com/systers/conference/register/RegisterActivity.java [hunk 2]
```diff
@@ -71,7 +71,7 @@ public void register() {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        if(AccountUtils.getRegisterVisited(this)){
+        if (AccountUtils.getRegisterVisited(this)) {
```
Comment (chhavip) [inline app/src/main/java/com/systers/conference/register/RegisterActivity.java]: same

File: app/src/main/java/com/systers/conference/register/RegisterActivity.java [hunk 3]
```diff
@@ -238,7 +238,7 @@ protected void onPostExecute(final Boolean success) {
                 AccountUtils.setEmail(activity, mEmail.getText().toString());
                 AccountUtils.setCompanyName(activity, mCompanyName.getText().toString());
                 AccountUtils.setCompanyRole(activity, mRole.getText().toString());
-                AccountUtils.setRegisterVisited(activity);
+                AccountUtils.setRegisterVisited(activity, true);
```
Comment (chhavip) [inline app/src/main/java/com/systers/conference/register/RegisterActivity.java]: same

**Comment by abs51295** [comment]
Actually @chhavip I did these changes to run UI tests itself. For example, I use setRegisterVisited in my RegisterActivity tests as follows: 

        @Override
        protected void beforeActivityLaunched() {
            AccountUtils.setRegisterVisited(context, false);
            super.beforeActivityLaunched();
        }

        @Override
        protected void afterActivityFinished() {
            AccountUtils.setRegisterVisited(context, true);
            super.afterActivityFinished();
        }

So, they are changes related to testing but not to this test. I added them because they were causing merge conflicts.

**Comment by riddhik84** [comment]
@abs51295 I would consider this as setup script and not the actual UI test since there is nothing being tested. The current tests are written as pre-requisite condition before running tests.

**Comment by abs51295** [comment]
Yes, @riddhik84 it is indeed a device state test and not a UI test. And yes, they are recommended to run before running all UI tests on the device. So what is your suggestion @riddhik84 ?

**Comment by abs51295** [comment]
@chhavip I have added the steps to run UI tests. Please check them 😄

**Comment by abs51295** [comment]
@chhavip The irrelevant changes have now been removed since all the tests have moved to a separate PR #58 and also I have rebased the PR for merging. So @MeepyMay @riddhik84 @chhavip please merge so that I can move to rebase the next one. 😄
